### PR TITLE
hotfix/APPEALS-46182

### PR DESCRIPTION
**Pull request by aferencz1987VA**
Resolves [APPEALS-46182](https://jira.devops.va.gov/browse/APPEALS-46182)

# Description
Due to added specialty case team distribution possibilities, more conditional logic was needed in order to make sure that blocking tasks parents were assigned correctly. 
## Acceptance Criteria
- [x] Code compiles correctly

## Testing Plan

1. Go to [Jira Issue](https://jira.devops.va.gov/browse/APPEALS-46182) or list them below

- [x] For feature branches merging into master: Was this deployed to UAT?

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec
- [ ] Jest
- [ ] Other

### CODE REVIEW

File: app/models/tasks/docket_switch/docket_switch_mail_task.rb [hunk 1]
```diff
@@ -29,7 +29,9 @@ def blocking?
     end
 
     def parent_if_blocking_task(parent_task)
-      if blocking? && !parent_task.appeal.distributed_to_a_judge?
+      return parent_task unless blocking?
+
+      if !parent_task.appeal.distributed_to_a_judge? || parent_task.appeal.specialty_case_team_assign_task?
```
Comment (raymond-hughes) [inline app/models/tasks/docket_switch/docket_switch_mail_task.rb:34]: Should this be 

`if !parent_task.appeal.distributed_to_a_judge? || !parent_task.appeal.specialty_case_team_assign_task?`

adding back in the ! for the specialty_case_team_assign_task

File: app/models/tasks/mail_task.rb [hunk 2]
```diff
@@ -45,7 +45,9 @@ def allow_creation?(_user = nil, _appeal = nil)
     end
 
     def parent_if_blocking_task(parent_task)
-      if blocking? && !parent_task.appeal.distributed_to_a_judge?
+      return parent_task unless blocking?
+
+      if !parent_task.appeal.distributed_to_a_judge? || parent_task.appeal.specialty_case_team_assign_task?
```
Comment (raymond-hughes) [inline app/models/tasks/mail_task.rb:50]: Same question here

File: spec/factories/appeal.rb [hunk 3]
```diff
@@ -476,6 +476,19 @@
       end
     end
```
Comment (raymond-hughes) [inline spec/factories/appeal.rb:478]: Do we need this factory anymore?

**Review by raymond-hughes — CHANGES_REQUESTED** [review]

File: app/models/tasks/docket_switch/docket_switch_mail_task.rb (hunk 1 above)
Comment (ajspotts) [inline app/models/tasks/docket_switch/docket_switch_mail_task.rb:34]: 👋 @aferencz1987VA hi my curiosity got the best of me 😄 

The bug [ticket](https://jira.devops.va.gov/browse/APPEALS-46182) references the `HearingRelatedMailTask` task [type](https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/models/tasks/hearing_related_mail_task.rb), which I see inherits from `MailTask` so I understand the changes to that model.

However it seems that `DocketSwitchMailTask` also directly inherits from `MailTask` as well.  Admittedly I'm a little bit rusty and could be missing some context but it looks like the two `parent_if_blocking_task` methods are identical.  So I guess my question would be do we even need `parent_if_blocking_task` here if it's written the exact same way in `MailTask`?

Comment (aferencz1987VA) [reply app/models/tasks/docket_switch/docket_switch_mail_task.rb:34]: That was the same question I had for Raymond. I would imagine that it does not need to exist. I didnt write the original method so I didnt want to remove it without approval.

File: spec/factories/appeal.rb (hunk 3 above)
Comment (aferencz1987VA) [reply spec/factories/appeal.rb:478]: Yes. While we have some stuff stubbed out, it still uses the factory.

**Review by raymond-hughes — APPROVED** [review]

**Review by raymond-hughes — APPROVED** [review]

**Review by raymond-hughes — APPROVED** [review]

**Pull request merged**